packages/zwave-js/src/lib/commandclass/MultilevelSwitchCC.ts

```diff
@@ -581,22 +621,26 @@ export class MultilevelSwitchCCSet extends MultilevelSwitchCC {
 	) {
 		super(driver, options);
 		if (gotDeserializationOptions(options)) {
-			// TODO: Deserialize payload
-			throw new ZWaveError(
-				`${this.constructor.name}: deserialization not implemented`,
-				ZWaveErrorCodes.Deserialization_NotImplemented,
-			);
+			validatePayload(this.payload.length >= 1);
+			this.targetValue = parseNumber(this.payload[0]);
+
+			if (this.version >= 2 && this.payload.length >= 2) {
+				this.targetValue = parseNumber(this.payload[0]);
```

Comment:
```suggestion
			if (this.payload.length >= 2) {
```

Reply:
Only the duration is optional and it is guaranteed to be at index 1 if the payload is long enough.